### PR TITLE
Company Billing method validation services

### PR DESCRIPTION
**Pull request by Kaian**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [ ] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

#### Description
- Validate Company billing method field when Brand has no Billing enable
- Ensure all Companies has billing method none if Brand Feature is being disabled

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Service\Company;
+
+use Ivoz\Core\Application\Service\EntityTools;
+use Ivoz\Provider\Domain\Model\Company\CompanyDto;
+use Ivoz\Provider\Domain\Model\Company\CompanyInterface;
+use Ivoz\Provider\Domain\Model\Feature\Feature;
+
+class ValidateBillingMethod implements CompanyLifecycleEventHandlerInterface
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php:10]: I think that SanitizeBillingMethod would be a better class name. I was expecting you to throw an exception on company "validation"

File: library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ];
+    }
+    /**
+     * @param FeaturesRelBrandInterface $featuresRelBrand
+     *
+     * @throws \DomainException
+     *
+     * @return void
+     */
+    public function execute(FeaturesRelBrandInterface $featuresRelBrand)
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php:32]:
```suggestion
    /**
     * @throws \DomainException
     */
    public function execute(FeaturesRelBrandInterface $featuresRelBrand): void
```

File: library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public const PRE_REMOVE_PRIORITY = self::PRIORITY_NORMAL;
+
+    public function __construct(
+        private CompanyRepository $companyRepository,
+    ) {
+    }
+
+
+    public static function getSubscribedEvents()
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php:19]:
```suggestion
    /** @return array<array-key, int> */
    public static function getSubscribedEvents()
```

File: library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+{
+    public const PRE_PERSIST_PRIORITY = self::PRIORITY_NORMAL;
+
+    public function __construct(
+        private EntityTools $entityTools,
+    ) {
+    }
+
+    public static function getSubscribedEvents()
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php:19]:
```suggestion
    /** @return array<array-key, int> */
    public static function getSubscribedEvents()
```

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php (hunk 4 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php:19]: No other service has that return hinting.

Don't forget the well known motto: _To' bien o to' mal, pero to' igua'_

File: library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php (hunk 1 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php]: Renamed! :+1:

File: library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php (hunk 4 above)
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Service/Company/ValidateBillingMethod.php]: Updated with the hint, I have added a task to update getSubscribedEvents in all services :+1:

File: library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        return [
+            self::EVENT_PRE_REMOVE => self::PRE_REMOVE_PRIORITY
+        ];
+    }
+
+    /**
+     * @throws \DomainException
+     */
+    public function execute(FeaturesRelBrandInterface $featuresRelBrand)
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/FeaturesRelBrand/AvoidBillingRemoval.php:29]:
```suggestion
    public function execute(FeaturesRelBrandInterface $featuresRelBrand): void
```

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**